src/config.js

```diff
@@ -34,7 +34,8 @@ export function applyConfigToArgv({
     if (!argv.hasOwnProperty(option)) {
       log.debug(`Ignoring configuration: ${option}=${configObject[option]} ` +
         'because this is an unknown option');
-      continue;
+      throw new UsageError(`The option ${option} is invalid.` +
+        ' Please fix your config and try again.');
```

Comment:
This error message should have the path to the config so that the user can find out which config they need to fix. This means `applyConfigToArgv()` will need to accept a filename argument.